.github/workflows/python-package.yaml

```diff
@@ -32,17 +32,21 @@ jobs:
 
       - uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
         with:
-          path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements*.txt') }}
+          path: ~/.cache/pypoetry
+          key: ${{ runner.os }}-poetry-${{ hashFiles('**/poetry.lock') }}
           restore-keys: |
-            ${{ runner.os }}-pip-
+            ${{ runner.os }}-poetry-
+
+      - name: Update PATH
+        run: 'echo "$HOME/.local/bin" >> "$GITHUB_PATH"'
 
       - name: Install dependencies
         env:
           CASS_DRIVER_NO_EXTENSIONS: theytaketoolongtobuild
         run: |
-          python --version
-          python -m pip install -r requirements.txt
+          pip install pipx
+          pipx install poetry==1.8.2
+          make venv
```

Comment:
pipx to install poetry? any reason not to use the curl script?

Reply:
I don't have a strong preference either way. For CI, their official docs recommend either one (although they do list `pipx` first on their official recommendations): https://python-poetry.org/docs/#ci-recommendations